cpp/tests/test_glct_secir.cpp

```diff
@@ -0,0 +1,525 @@
+/* 
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: Lena Ploetzke
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "glct_secir/model.h"
+#include "glct_secir/parameters.h"
+#include "memilio/config.h"
+#include "memilio/utils/time_series.h"
+#include "memilio/utils/logging.h"
+#include "memilio/epidemiology/uncertain_matrix.h"
+#include "memilio/compartments/simulation.h"
+#include "memilio/math/eigen.h"
+#include "load_test_data.h"
+
+#include <gtest/gtest.h>
+#include "boost/numeric/odeint/stepper/runge_kutta_cash_karp54.hpp"
+
+// Test if the function eval_right_hand_side() is working using a hand calculated result.
+TEST(TestGLCTSecir, testEvalRightHandSide)
+{
+    // Define model.
+    // Chose more than one subcompartment for all compartments except S, R, D so that the function is correct for all selections.
```

Comment:
```suggestion
    // Choose more than one subcompartment for all compartments except S, R, D to check that the function eval_tight_hand_side() is correct for all selections.
```
Is this what you mean?